client/src/components/SessionForm.jsx

```diff
@@ -0,0 +1,221 @@
+import React, { useState } from 'react';
+import { Modal, Form, Input, TextArea, Radio } from 'semantic-ui-react';
+
+import 'semantic-ui-css/semantic.min.css';
+import { addSession } from '../utils/apiWrapper';
+import '../css/SessionForm.scss';
+
+function SessionForm(props) {
+  const { button } = props;
+  const [open, setOpen] = useState(false);
+  const [isLater, setIsLater] = useState(false);
+  const [courseCode, setCourseCode] = useState('');
+  const [courseNumber, setCourseNumber] = useState();
+  const [courseSuffix, setCourseSuffix] = useState('');
+  const [location, setLocation] = useState('');
+  const [attendees, setAttendees] = useState('');
+  const [notes, setNotes] = useState('');
+  const [date, setDate] = useState();
+  const [startTime, setStartTime] = useState();
+  const [endTime, setEndTime] = useState();
+  const [endTimeDefined, setEndTimeDefined] = useState(false);
+
+  const processFormAndSubmit = async () => {
+    const creatorID = 'jaskdlfioawh';
+    const course = courseCode + courseNumber + courseSuffix;
+    const attendeeArray = attendees.split(',');
+
+    const defaultTimeout = 43200;
+    const millisecondsInDay = 86400000;
+    const active = !isLater;
+    const processedStartDate = active ? 0 : date.split('-'); //2021-11-17
+    const processedStartTime = active ? 0 : startTime.split(':');
+    const laterStart = active
+      ? 0
+      : new Date(
+          parseInt(processedStartDate[0]),
+          parseInt(processedStartDate[1]),
+          parseInt(processedStartDate[2]),
+          parseInt(processedStartTime[0]),
+          parseInt(processedStartTime[1]),
+          0,
+          0,
+        );
+    const processedStart = active ? new Date() : laterStart;
+    const startSeconds = processedStart.getTime() / 1000;
+
+    const processedEndTime = endTimeDefined ? endTime.split(':') : 0;
+    const end = endTimeDefined
+      ? new Date(
+          processedStart.getFullYear(),
+          processedStart.getMonth(),
+          processedStart.getDate(),
+          processedEndTime[0],
+          processedEndTime[1],
+          0,
+          0,
+        )
+      : 0;
+    const endSeconds = endTimeDefined ? end.getTime() / 1000 : 0;
+    const timeoutTry = endTimeDefined
+      ? endSeconds - startSeconds
+      : defaultTimeout;
+    const timeout =
+      timeoutTry < 0
+        ? new Date(end.getTime() + millisecondsInDay).getTime() / 1000 -
+          startSeconds
+        : timeoutTry;
+    console.log(startSeconds);
+
+    const sessionData = {
+      creator: creatorID,
+      class: course,
+      location: location,
+      attendees: attendeeArray,
+      notes: notes,
+      active: !isLater,
+      startTime: startSeconds,
+      timeout: timeout,
+    };
+    console.log(sessionData);
+    await addSession(sessionData);
```

Comment:
A next step (no need to add this to this PR): if creating the session is successful, close the modal. Otherwise, show an error message and ask them to edit/resubmit the form.